fileio/ft_read_data.m

```diff
@@ -1026,11 +1026,33 @@
     % recordings. The code currently concatenates these trials.
     % We could set this up as separate "trials" later.
     % We could probably add "boundary events" in EEGLAB later
-    dat = zeros(0,size(hdr.orig.Segments(1).chName,2));
-    for segment=1:size(hdr.orig.Segments,2)
-      range = [1 hdr.orig.Segments(segment).sampleCount];
-      datseg = read_nervus_data(hdr.orig,segment, range, chanindx);
-      dat = cat(1,dat,datseg);
+    
+    %Fieldtrip can't handle multiple sampling rates in a data block
+    %We will get only the data with the most frequent sampling rate
+    
+        
+    targetNumberOfChannels = hdr.orig.targetNumberOfChannels;
+    targetSampleCount = hdr.orig.targetSampleCount;
+    
+    dat = zeros(targetSampleCount,targetNumberOfChannels);
```

Comment:
I haven't thought this through, but I am surprised that this (used to) work(s). Typically, a low-level reading function returns a chan x time matrix, with time in the columns. It seems that here the data is constructed as a time x chan matrix. Where does the matrix transposition occur?

Reply:
The array layout is indicated in ft_read_data.m line 1057: 
`dimord = 'samples_chans';`
If I try to switch things around, it messes up the test code.

Suggest leaving it be.

Reply:
OK, it seems to work, so probably, ft_read_data is clever enough to transpose the data if needed.